File: src/main/java/com/example/demo/Preference.java
package com.example.demo;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;


@ConfigurationProperties(prefix = "preference")
@Data
public class Preference {

    private PageProperty topPage;
    private PageProperty detailPage;

    @Data
    public static class PageProperty {
        private int listLength;
        private boolean checked;
    }
}

File: src/main/java/com/example/demo/Preference.kt
package com.example.demo

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@ConfigurationProperties(prefix = "preference")
@Component
class Preference(var topPage: PageProperty?, var detailPage: PageProperty?)

class PageProperty(var listLength: Int, var isChecked: Boolean)
